src/main/java/com/amazonaws/services/kinesis/multilang/MultiLangDaemonConfig.java

```diff
@@ -55,65 +58,103 @@
+     * @param configurator   A configurator to use.
+     * @throws IOException              Thrown when the properties file can't be accessed.
      * @throws IllegalArgumentException Thrown when the contents of the properties file are not as expected.
      */
-    public MultiLangDaemonConfig(String propertiesFile,
-            ClassLoader classLoader,
+    public MultiLangDaemonConfig(String propertiesFile, ClassLoader classLoader,
             KinesisClientLibConfigurator configurator) throws IOException, IllegalArgumentException {
         Properties properties = loadProperties(classLoader, propertiesFile);
         if (!validateProperties(properties)) {
-            throw new IllegalArgumentException("Must provide an executable name in the properties file, "
-                    + "e.g. executableName = sampleapp.py");
+            throw new IllegalArgumentException(
+                    "Must provide an executable name in the properties file, " + "e.g. executableName = sampleapp.py");
         }
 
         String executableName = properties.getProperty(PROP_EXECUTABLE_NAME);
         String processingLanguage = properties.getProperty(PROP_PROCESSING_LANGUAGE);
+        ClientConfiguration clientConfig = buildClientConfig(properties);
+
+        kinesisClientLibConfig = configurator.getConfiguration(properties).withKinesisClientConfig(clientConfig)
+                .withCloudWatchClientConfig(clientConfig).withDynamoDBClientConfig(clientConfig);
 
-        kinesisClientLibConfig = configurator.getConfiguration(properties);
         executorService = buildExecutorService(properties);
-        recordProcessorFactory = new MultiLangRecordProcessorFactory(executableName, executorService, kinesisClientLibConfig);
+        recordProcessorFactory = new MultiLangRecordProcessorFactory(executableName, executorService,
+                kinesisClientLibConfig);
 
         LOG.info("Running " + kinesisClientLibConfig.getApplicationName() + " to process stream "
                 + kinesisClientLibConfig.getStreamName() + " with executable " + executableName);
         prepare(processingLanguage);
     }
 
+    private ClientConfiguration buildClientConfig(Properties properties) {
+        ClientConfiguration clientConfig = new ClientConfiguration();
+        String proxyHost = null;
+        int proxyPort = 0;
+
+        if (properties.getProperty("http.proxyHost") != null) {
+            LOG.debug("Getting proxy info from properties file.");
+
+            proxyHost = properties.getProperty("http.proxyHost");
+            proxyPort = Integer.parseInt(properties.getProperty("http.proxyPort"));
+        } else if (System.getProperty("http.proxyHost") != null) {
+            LOG.debug("Getting proxy info from java system properties");
+
+            proxyHost = System.getProperty("http.proxyHost");
+            proxyPort = Integer.parseInt(System.getProperty("http.proxyPort"));
+        } else if (System.getenv("http_proxy") != null) {
```

Comment:
Can you change the property to upper case HTTP_PROXY, when reading from System properties to be inline with the example here 

https://docs.aws.amazon.com/cli/latest/userguide/cli-http-proxy.html